gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -279,14 +272,20 @@ function GM:HUDDrawTargetID()
 	-- only add onscreen infos when the entity isn't the local player
 	if ent == client then return end
 
-	ent = hook.Run("TTTModifyTargetedEntity", ent, distance) or ent
+	local changedEnt = hook.Run("TTTModifyTargetedEntity", ent, distance)
+
+	if IsValid(changedEnt) then
```

Comment:
Just check `if changedEnt then`. It doesn't matter whether it's valid or not. If a script is going to change it, you shouldn't interrupt it.